### ISSUE TITLE
Use of terms --flag and --option

### ISSUE THREAD

**Issue by edwardxml** [rclone/rclone#4756]
#### What is your current rclone version (output from `rclone version`)?

Current rclone documentation at 10 - 11 - 2020

#### What problem are you are trying to solve?

There is some inconsistency in application relating to `--flags` and `--options`. `rclone help` uses 'flags' whilst the documentation uses both interchangeably - sometimes on the same page eg https://rclone.org/filtering/

I tend to use 'flags' colloquially and especially to rather binary situations (reversing default behaviour) and 'options' for situations where more varied values can be applied to a command line setting. There don't seem to be many, if any, references to them as 'modifiers' or 'arguments' but I have seen reference to 'parameters' by a leading contributor.

#### How do you think rclone should be changed to solve that?

Adopt a standard terminology that can be moved to over time in documentation for the use of 'flag'/'option' on command lines.

**Comment by animosity22** [comment]
If you'd like to submit a PR to update those, please have at it.

**Comment by edwardxml** [comment]
Which way should they generally be described? I had been working on the filters docs page and this was an issue that came up there. I'm tempted to go with 'options' but the `rclone help` and a lot of the forum use 'flags'

**Comment by ncw** [comment]
The library rclone uses to parse them is called `pflag` which is probably where most of the `flag` usage comes in! Wheras the C libary is called `getopt`.

Doing a bit of reading `flag` does tend to imply a boolean value whereas `option` doesn't. So `flag` is a subset of `option`.

Interestingly the Go [standard library to parse flags/options](https://golang.org/pkg/flag/) does not mention the word option. The Go team are really hot on terminology and the core members include a number of very well known unix people!

Note that rclone has a command `rclone help flags thing` which will find all ~~flags~~options with `thing` in them.

"Command line options" has about 500M hits on google wheras "Command line flags" only has 50M FWIW.

> There don't seem to be many, if any, references to them as 'modifiers' or 'arguments' but I have seen reference to 'parameters' by a leading contributor.

`modifiers` is a term I've heard but it isn't common. `arguments` has a specific meaning - it is either the entire command line, or the parts of the command line which aren't part of an `option`. `parameters` is more of a programming term I think.

So, in summary

- `flag` is good for binary options
- `option` should be used in preference for options which take arguments
- `option` is a superset of `flag` and could be used everywhere.

So I think there are some incorrect uses of `flag` in the filtering docs.

We could change all uses of `flag` to `option` - that might involve changing rather a lot of docs though!

**Comment by edwardxml** [comment]
>The library rclone uses to parse them is called pflag which is probably where most of the flag...

That makes a lot of sense.

>rclone help flags thing

A good reason to go with flags entirely, as is my favourite error message when I forget to quote things and rclone complains about too many non flag arguments. `rclone help` itself refers to flags not options. My intention is that the documentation is itself consistent and consistent with the binary. I'm thinking there are lots of reference to flags in messages, comments and variable names in the binary and virtually none to options. Changing them wouldn't be an option, especially as changing messages could break backwards compatibility in scripts parsing rclone logs.

asdffdsa often refers to all 'options' as 'flags' (especially when encouraging OP's to use less of them) and everyone understands the meaning. You tend to use the phrase 'flags' for all 'options' in the forum, and again the meaning is clear. Animosity022 refers to the things without often committing to a name - ie "use --something-or-other".

>We could change all uses of flag to option

I am increasingly contemplating the opposite and changing all the `option` references to `flag`. It isn't as if this is objectively wrong and it is as you point out consistent with the go documentation.

**Comment by edwardxml** [comment]
Everywhere I look in the code and discussion the emphasis is on flags not options even when supplying complex values so I am closing this issue.

**Comment by ncw** [comment]
OK seems reasonable :-) I'm pretty sure [Rob Pike](https://en.wikipedia.org/wiki/Rob_Pike) wrote the go flag library and if he's happy with the word flag, that is good enough for me!
